src/object/cli_obj.c

```diff
@@ -4760,9 +4760,20 @@ obj_comp_cb(tse_task_t *task, void *data)
 						obj_auxi->tx_uncertain = 1;
 					else
 						obj_auxi->nvme_io_err = 1;
-				} else if (task->dt_result != -DER_NVME_IO) {
-					/* Don't retry update for CSUM & UNCERTAIN errors */
-					obj_auxi->io_retry = 0;
+				} else {
+					if (task->dt_result == -DER_CSUM) {
+						/** Retry a few times on checksum error on update */
+						if (!obj_auxi->csum_retry &&
```

Comment:
"!obj_auxi->csum_retry" seems can be removed as it is cleared at above L4689 so it is always true.